src/main/java/com/laytonsmith/core/ParseTree.java

```diff
@@ -389,4 +393,29 @@ public Target getTarget() {
 		}
 	}
 
+	/**
+	 * Returns the CClassType of this node. For constants, this is just the type, for variables, this is the defined
+	 * type, and for functions, this is the return type of the function. For many value, the type will be AUTO. The
+	 * type may be CNull for literal nulls, but will never be java null.
+	 * @return
+	 */
+	public CClassType getType(Environment env) {
+		if(isConst()) {
+			return getData().typeof();
+		} else if(getData() instanceof IVariable ivar) {
+			return ivar.getDefinedType();
+		} else if(getData() instanceof CFunction cf) {
+			List<CClassType> argTypes = new ArrayList<>();
+			List<Target> argTargets = new ArrayList<>();
+			Set<ConfigCompileException> exceptions = new HashSet<>();
+			for(ParseTree child : getChildren()) {
+				argTypes.add(child.getType(env));
+				argTargets.add(child.getTarget());
+			}
+			return cf.getCachedFunction().getReturnType(getTarget(), argTypes, argTargets, env, exceptions);
```

Comment:
This doesn't do anything with possible exceptions. Not sure if this is intended to be added later.

Reply:
Hmm, that's a good point. In general, this shouldn't throw an exception, because of the semi-declarative nature, but for malformed functions, that would themselves be giving out compile errors, that's a good point, and I'm not sure what the behavior would be.

Reply:
The function signature doesn't currently allow for exceptions to be thrown, so I'm just going to leave this as won't fix for now. However, I suspect that this will crop up again later, but it will be easier to understand the expected behavior then, and the solution here might be more obvious.